routers/repo/repo.go

```diff
@@ -268,6 +268,31 @@ func Action(ctx *context.Context) {
 	ctx.Redirect(redirectTo)
 }
 
+// RedirectDownload return a file based on the the following infos:
+func RedirectDownload(ctx *context.Context) {
+	var (
+		vTag     = ctx.Params("vTag")
+		fileName = ctx.Params("fileName")
+	)
+	tagNames := []string{vTag}
+	curRepo := ctx.Repo.Repository
+	releases, err := models.GetReleasesByRepoIDAndNames(curRepo.ID, tagNames)
+	if err != nil {
+		ctx.Handle(500, "RedirectDownload -> Release not found", err)
+		return
+	}
+
+	if len(releases) == 1 {
+		release := releases[0]
+		att, err := models.GetAttachmentByReleaseIDFileName(release.ID, fileName)
+		if err != nil {
+			ctx.Handle(500, "RedirectDownload -> Attachment not found", err)
+			return
+		}
+		ctx.Redirect(setting.AppSubURL + "/attachments/" + att.UUID)
+	}
+}
```

Comment:
End the function with `ctx.Handle(404, "RedirectDownload", nil)` otherwise you get a 500 error if the release doesn't exist :)